py-polars/tests/unit/dataframe/test_df.py

```diff
@@ -3463,23 +3465,28 @@ def test_format_empty_df() -> None:
 
 
 def test_deadlocks_3409() -> None:
-    assert (
-        pl.DataFrame({"col1": [[1, 2, 3]]})
-        .with_columns(
-            [
-                pl.col("col1").list.eval(
-                    pl.element().apply(lambda x: x, return_dtype=pl.Int64)
-                )
-            ]
-        )
-        .to_dict(False)
-    ) == {"col1": [[1, 2, 3]]}
+    with catch_warnings():
```

Comment:
could we use `pytest.warns` instead of turning the warning off please? this way, if something changed and the warning unexpectedly wasn't raised, we'd be immediately notified